core/foundation/inc/RtypesCore.h

```diff
@@ -34,91 +45,91 @@ class TRootIOCtor;
-typedef short          Version_t;   //Class version identifier (short)
-typedef const char     Option_t;    //Option string (const char)
-typedef int            Ssiz_t;      //String size (int)
-typedef float          Real_t;      //TVector and TMatrix element type (float)
-typedef long long          Long64_t; //Portable signed long integer 8 bytes
-typedef unsigned long long ULong64_t;//Portable unsigned long integer 8 bytes
+typedef float          Float_t;     ///< Float 4 bytes (float) \deprecated Consider replacing with `float`.
+typedef float          Float16_t;   ///< Float 4 bytes in memory, written to disk as a 3 bytes (24-bits, not 16-bits) float with a truncated (up-to) 16-bit mantissa (12-bit by default in memory), and (7+1)-bits of exponent \deprecated Consider (non-equivalent) alternatives in `<cstdfloat>`. \warning Do not mix this with a half-float such as std::float16_t
+typedef double         Double_t;    ///< Double 8 bytes \deprecated Consider replacing with `double`.
+typedef double         Double32_t;  ///< Double 8 bytes in memory, written to disk as a 3 bytes (24-bits, not 32-bits) float with a truncated (up-to) 16-bit mantissa (24-bit by default in memory), and (7+1)-bits of exponent \deprecated Consider (non-equivalent) alternatives in `<cstdfloat>`. \warning Do not mix this with a single-precision float std::float32_t
+typedef long double    LongDouble_t;///< Long Double (not portable) \deprecated Consider replacing with `long double`.
+typedef char           Text_t;      ///< General string (char) \deprecated Consider replacing with `char`.
+typedef bool           Bool_t;      ///< Boolean (0=false, 1=true) (bool) \deprecated Consider replacing with `bool`.
+typedef unsigned char  Byte_t;      ///< Byte (8 bits) (unsigned char) \deprecated Consider replacing with `unsigned char` or `std::byte`.
+typedef short          Version_t;   ///< Class version identifier (short) \deprecated Consider replacing with `short` or `std::int16_t`.
+typedef const char     Option_t;    ///< Option string (const char) \deprecated Consider replacing with `const char`.
+typedef int            Ssiz_t;      ///< String size (int) \deprecated Consider replacing with `int`.
+typedef float          Real_t;      ///< TVector and TMatrix element type (float) \deprecated Consider replacing with `float`.
+typedef long long          Long64_t; ///< Portable signed long integer 8 bytes \deprecated Consider replacing with `long long` or `std::int64_t`.
+typedef unsigned long long ULong64_t;///< Portable unsigned long integer 8 bytes \deprecated Consider replacing with `unsigned long long` or `std::uint64_t`.
 #ifdef _WIN64
-typedef long long      Longptr_t;   //Integer large enough to hold a pointer
-typedef unsigned long long ULongptr_t;  //Unsigned integer large enough to hold a pointer
+typedef long long      Longptr_t;     ///< Integer large enough to hold a pointer \deprecated Consider replacing with `long long` on Windows.
```

Comment:
> ///< .... on Windows.
> ///< .... on UNIX.

This is confusing.  The purpose of the typedef is to avoid having the user use a different type on different platform (where user includes our own code).

Reply:
But in that case, the alternative data type is not portable. Conclusion, we should not deprecate these ones?

Reply:
If the alternative is not valid on all platform for 'all' circumstances, I don't think we should deprecate.